### DISCUSSION TITLE
[monorepo] I can't import my components in its own stories, after migrating our monorepo from yarn to pnpm

### DISCUSSION THREAD

**Discussion by nirbG** [pnpm/pnpm#6007] [Q&A]
Hey guys,
My teams and I moved from yarn to pnpm. but I am facing issues to import our components in our stories.

to sum up, we have decided to create a design system using Lit and storybook.
We have a dedicated npm package for storybook and a dedicated npm package for each components.

```
|___ MONOREPO
   |___ apps
   |  |__ [...]
   |
   |___ packages
     |___ micro-front-ends
     |   |__ [...]
     |
     |___ UI library
        |___ storybook
        |   |__.stoybook
        |   |___docs
        |   |___package.json
        |
        |___ components
           |___ badge
           |    |___ stories
           |    |  |__ badge.stories.ts
           |    |__ badge.ts
           |    |__ package.json
           |
           |___ button
           |    |___ stories
           |    |  |__ button.stories.ts
           |    |__ button.ts
           |    |__ package.json
           |
           |___ text-field
```

Here is a example of our stories. in this case is the `badge.stories.ts`

```
import {TemplateResult} from 'lit';
import {html} from 'lit/static-html.js';

import '@blabla/wc-storybook-utils';

import '@blabla/wc-svg-icon';
import {unsafe_fui_svg_icon} from '@blabla/wc-svg-icon';

import '@blabla/wc-badge';
import {unsafe_fui_badge} from '@blabla/wc-fui-badge';

export default {
  title: 'Components/Badges',
  argTypes: {..}
}

export const Sandbox = (controls): TemplateResult => {
  return html`
    <story-sandbox>
      <${unsafe_fui_badge}
        .color=${'primary'}
        .clickable=${true}>
        <${unsafe_svg_icon}slot="prefix" .size=${2}.svg=${'add'}></${unsafe_fui_svg_icon}>
          ${controls.label}
      </${unsafe_fui_badge}>
    </story-sandbox>
  `;
};
```

With yarn, I was able to import @blabla/wc-badge from its own npm package.
it worked because it was a phantom dependencies.
But since we moved to pnpm it doesn't works anymore. Because, if I don't specify this npm package in the package.json I won't be able to import it or it not even added to the node module at all.

For the moment I tried :
- To replace npm package name from the import by the name of the file but it doesn't works, 
   - import '../Badge';
- I tried to play with the .npmrc but I never found the right solution
- I tried to replace the package name by the path to the dist of the component. it works but I really don't like this solution
   - import '../dist';
- I tried to search how people use storybook and pnpm but I don't find a lot of example or docuementation

Has anyone run into the same problem ?
Or do you know how to configure pnpm and storybook for this kind of architecture ?

**Comment by weyert** [comment]
So if I understand you correctly you try to import a file from it's own package in the stories file?

**Reply by nirbG** [reply]
yeah,
exactly.

I had to do it that way because with Lit, I had issue to import the component with the relative path. that's why, I had to use this workaround and it worked because I was using yarn.

I just figure it out why I couldn't do it previously. 
I didn't know that storybook use babel under the hood because they were using their own babel config.
but in the new beta version we have to create our won babel config. I found why I had this error before.

https://lit.dev/docs/components/decorators/#using-decorators-with-typescript-and-babel
